crates/packager/src/package/app/mod.rs

```diff
@@ -154,12 +154,9 @@ pub(crate) fn package(ctx: &Context) -> crate::Result<Vec<PathBuf>> {
         // notarization is required for distribution
         match config
             .macos()
-            .map(|m| {
-                m.notarization_credentials
-                    .clone()
-                    .ok_or_else(|| crate::Error::MissingNotarizeAuthVars)
-            })
-            .unwrap_or_else(codesign::notarize_auth)
+            .and_then(|m| m.notarization_credentials.clone())
+            .ok_or(crate::Error::MissingNotarizeAuthVars)
+            .or_else(|_| codesign::notarize_auth())
```

Comment:
No need to add an error variant here, since if `notarization_credentials` is not set, it will fallback to `codesign::notarize_auth` and that one returns `crate::Error::MissingNotarizeAuthVars` if the needed environment variables are not set. 
```suggestion
            .and_then(|m| m.notarization_credentials.clone())
            .unwrap_or_else(codesign::notarize_auth)
```

Reply:
well I only added that in order to make things compile properly, as this line
```rust
 .and_then(|m| m.notarization_credentials.clone())
 ```
"emits" an `Option<MacOsNotarizationCredentials>` type, while the next line
```rust
.unwrap_or_else(codesign::notarize_auth)
```
"emits" an `Result<MacOsNotarizationCredentials>`.

So the two choices we have are:
1. convert the second line to `codesign::notarize_auth().ok()`, which loses the error info.
2. add the error variant as shown above, which doesn't lose the error context.

Which would you prefer?

Reply:
I see, then your changes is fine, thanks